buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-					bpPlan := buildpack.Plan{
-						Entries: []buildpack.Require{
-							{
-								Name:    "some-deprecated-bp-dep",
-								Version: "v1", // top-level version is deprecated in buildpack API 0.3
-							},
-							{
-								Name:    "some-deprecated-bp-replace-version-dep",
-								Version: "some-version-orig", // top-level version is deprecated in buildpack API 0.3
-							},
-							{
-								Name:     "some-dep",
-								Metadata: map[string]interface{}{"version": "v1"},
-							},
-							{
-								Name:     "some-replace-version-dep",
-								Metadata: map[string]interface{}{"version": "some-version-orig"},
-							},
-							{
-								Name: "some-unmet-dep",
-							},
-						},
-					}
-
-					h.Mkfile(t,
-						"[[entries]]\n"+
-							`name = "some-deprecated-bp-dep"`+"\n"+
-							`version = "v1"`+"\n"+
-							"[[entries]]\n"+
-							`name = "some-deprecated-bp-replace-version-dep"`+"\n"+
-							`version = "some-version-new"`+"\n"+
-							"[[entries]]\n"+
-							`name = "some-dep"`+"\n"+
-							"[entries.metadata]\n"+
-							`version = "v1"`+"\n"+
-							"[[entries]]\n"+
-							`name = "some-replace-version-dep"`+"\n"+
-							"[entries.metadata]\n"+
-							`version = "some-version-new"`+"\n",
-						filepath.Join(appDir, "build-plan-out-A-v1.toml"),
-					)
-
-					br, err := bpTOML.Build(bpPlan, config, mockEnv)
-					if err != nil {
-						t.Fatalf("Unexpected error:\n%s\n", err)
-					}
-
-					if s := cmp.Diff(br, buildpack.BuildResult{
-						LaunchBOM: []buildpack.BOMEntry{
-							{
-								Require: buildpack.Require{
-									Name:     "some-deprecated-bp-dep",
-									Metadata: map[string]interface{}{"version": "v1"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-							{
-								Require: buildpack.Require{
-									Name:     "some-deprecated-bp-replace-version-dep",
-									Metadata: map[string]interface{}{"version": "some-version-new"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-							{
-								Require: buildpack.Require{
-									Name:     "some-dep",
-									Metadata: map[string]interface{}{"version": "v1"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-							{
-								Require: buildpack.Require{
-									Name:     "some-replace-version-dep",
-									Metadata: map[string]interface{}{"version": "some-version-new"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-						},
-						Labels: nil,
-						MetRequires: []string{
-							"some-deprecated-bp-dep",
-							"some-deprecated-bp-replace-version-dep",
-							"some-dep",
-							"some-replace-version-dep",
-						},
-						Processes: nil,
-						Slices:    nil,
-					}); s != "" {
-						t.Fatalf("Unexpected:\n%s\n", s)
-					}
-				})
-
-				it("should convert top level version to metadata.version in the bom", func() {
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L1057-L1100